tests/runtime/in_event_test.c

```diff
@@ -0,0 +1,39 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+#include <fluent-bit.h>
+#include "flb_tests_runtime.h"
+
+void flb_test_input_event()
+{
+    int i;
+    int ret;
+    int bytes;
+    char p[100];
+    flb_ctx_t *ctx;
+    int in_ffd;
+    int out_ffd;
+    int filter_ffd;
+
+    ctx = flb_create();
+
+    in_ffd = flb_input(ctx, (char *) "event_test", NULL);
+    TEST_CHECK(in_ffd >= 0);
+
+    out_ffd = flb_output(ctx, (char *) "null", NULL);
+    TEST_CHECK(out_ffd >= 0);
+    flb_output_set(ctx, out_ffd, "match", "test", NULL);
+
+    ret = flb_start(ctx);
+    TEST_CHECK(ret == 0);
+
+    sleep(8);
+
+    flb_stop(ctx);
+    flb_destroy(ctx);
+}
```

Comment:
This test can't detect below error.

I modified longer interval like below.
```diff
iff --git a/plugins/in_event_test/event_test.c b/plugins/in_event_test/event_test.c
index 5058926..cb18214 100644
--- a/plugins/in_event_test/event_test.c
+++ b/plugins/in_event_test/event_test.c
@@ -304,7 +304,7 @@ static int cb_event_test_init(struct flb_input_instance *ins,
 
     /* socket client: connect to socket server to trigger the event */
     ret = flb_input_set_collector_time(ins, cb_collector_server_client,
-                                       CALLBACK_TIME * 2, 0, config);
+                                       CALLBACK_TIME * 200, 0, config);
     if (ret < 0) {
         return -1;
     }
```

However it didn't detect error.
```
$ bin/flb-rt-in_event_test 
Test event_test...                              [2022/07/03 11:10:46] [ info] [fluent bit] version=1.9.5, commit=eb8dd34c6b, pid=57103
[2022/07/03 11:10:46] [ info] [storage] version=1.2.0, type=memory-only, sync=normal, checksum=disabled, max_chunks_up=128
[2022/07/03 11:10:46] [ info] [cmetrics] version=0.3.1
[2022/07/03 11:10:46] [ info] [input:event_test:event_test.0] thread instance initialized
[2022/07/03 11:10:46] [ info] [sp] stream processor started
[2022/07/03 11:10:46] [ info] [input:event_test:event_test.0] pre run OK
[2022/07/03 11:10:46] [ info] [output:null:null.0] worker #0 started
[2022/07/03 11:10:48] [ info] [input:event_test:event_test.0] [OK] collector_time
[2022/07/03 11:10:48] [ info] [input:event_test:event_test.0] [OK] collector_fd
[2022/07/03 11:10:54] [ warn] [engine] service will shutdown in max 5 seconds
[2022/07/03 11:10:54] [ info] [input:event_test:event_test.0] [OK] engine has paused the plugin
[2022/07/03 11:10:55] [ info] [engine] service has stopped (0 pending tasks)
[2022/07/03 11:10:55] [ info] [input:event_test:event_test.0] [OK] engine has paused the plugin
[2022/07/03 11:10:55] [ info] [input:event_test:event_test.0] unit test #0 'collector time' succeeded
[2022/07/03 11:10:55] [ info] [input:event_test:event_test.0] unit test #1 'collector fd_event' succeeded
[2022/07/03 11:10:55] [error] [input:event_test:event_test.0] unit test #2 'collector fd_server | socket' failed
[2022/07/03 11:10:55] [ info] [input:event_test:event_test.0] unit test #3 'plugin paused from engine' succeeded
[2022/07/03 11:10:55] [error] [input:event_test:event_test.0] unit test #4 'plugin resumed from engine' failed
[2022/07/03 11:10:55] [ info] [output:null:null.0] thread worker #0 stopping...
[2022/07/03 11:10:55] [ info] [output:null:null.0] thread worker #0 stopped
[ OK ]
SUCCESS: All unit tests have passed.
```

Reply:
thanks, good catch.

I have fixed the exit and now the process exit with `EXIT_FAILURE`:

```
Fluent Bit v1.9.6
* Copyright (C) 2015-2022 The Fluent Bit Authors
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2022/07/07 00:11:28] [ info] [fluent bit] version=1.9.6, commit=deb25643e7, pid=83650
[2022/07/07 00:11:28] [ info] [storage] version=1.2.0, type=memory-only, sync=normal, checksum=disabled, max_chunks_up=128
[2022/07/07 00:11:28] [ info] [cmetrics] version=0.3.5
[2022/07/07 00:11:28] [ info] [input:event_test:event_test.0] thread instance initialized
[2022/07/07 00:11:28] [ info] [output:stdout:stdout.0] worker #0 started
[2022/07/07 00:11:28] [ info] [sp] stream processor started
[2022/07/07 00:11:28] [ info] [input:event_test:event_test.0] pre run OK
[2022/07/07 00:11:32] [ info] [input:event_test:event_test.0] [OK] collector_server_socket
[2022/07/07 00:11:32] [ info] [input:event_test:event_test.0] test pause/resume in 5 seconds...
[2022/07/07 00:11:37] [ info] [input:event_test:event_test.0] [OK] engine has resumed the plugin
[2022/07/07 00:11:37] [ warn] [engine] service will shutdown in max 5 seconds
[2022/07/07 00:11:37] [ info] [input:event_test:event_test.0] [OK] engine has paused the plugin
[2022/07/07 00:11:37] [ info] [input:event_test:event_test.0] [OK] engine has paused the plugin
[2022/07/07 00:11:38] [ info] [engine] service has stopped (0 pending tasks)
[2022/07/07 00:11:38] [ info] [input:event_test:event_test.0] [OK] engine has paused the plugin
[2022/07/07 00:11:38] [error] [input:event_test:event_test.0] unit test #0 'collector time' failed
[2022/07/07 00:11:38] [error] [input:event_test:event_test.0] unit test #1 'collector fd_event' failed
[2022/07/07 00:11:38] [ info] [input:event_test:event_test.0] unit test #2 'collector fd_server | socket' succeeded
[2022/07/07 00:11:38] [ info] [input:event_test:event_test.0] unit test #3 'plugin paused from engine' succeeded
[2022/07/07 00:11:38] [ info] [input:event_test:event_test.0] unit test #4 'plugin resumed from engine' succeeded
root@8096f83a0ec1:/coding/fluent-bit/build# echo $?
1
```